src/commands/deploy/pipeline.ts

```diff
@@ -24,23 +23,24 @@ export default class DeployPipeline extends PromoteCommand<typeof SfCommand> {
   public static readonly state = 'beta';
 
   public async run(): Promise<PromotePipelineResult> {
-    // TODO Timeout case
-    if (this.flags.async) {
-      // TODO Get the aorId
-      const aorId = 'TODO';
-      await DeployPipelineCache.set(aorId, {});
-    }
     return this.executePromotion();
   }
 
   /**
-   * Computes the target stage Id for deployOnly promotions. If the given stage has a previous stage
-   * then the target stage is the previous stage. If not the it means this is the
-   * first stage of the pipeline and target stage Id = Approved.
+   * Computes the source stage Id for deployOnly promotions. If the given stage has a previous stage
+   * then the source stage is the previous stage. If not the it means this is the
+   * first stage of the pipeline and source stage Id = Approved.
    */
-  protected computeTargetStageId(pipelineStage: PipelineStage): void {
-    this.targetStageId = pipelineStage?.sf_devops__Pipeline_Stages__r
+  protected computeSourceStageId(pipelineStage: PipelineStage): void {
+    this.sourceStageId = pipelineStage?.sf_devops__Pipeline_Stages__r
       ? pipelineStage.sf_devops__Pipeline_Stages__r.records[0].Id
       : 'Approved';
   }
+
+  /**
+   * Adds a promote option so it performs an undeployed only promotion
+   */
+  protected addPromoteOptions(): void {
+    this.deployOptions = { ...this.deployOptions, undeployedOnly: true };
+  }
```

Comment:
Minor comment about computeTargetStage and addPromoteOptions.  It is a bit safer to have the abstract class be the only one that deals with setting the member variables (and then they can be private).  In this case it might be a bit safer/cleaner if you had

computeTargetStageId(pipelineStage: PipelineStage): string

and

addPromoteOption(options: Partial\<PromoteOptions\>)

Reply:
Good to know, I changed it so the abstract class is the one that updates its members.